src/main/java/seedu/address/logic/parser/ScheduleCommandParser.java

```diff
@@ -0,0 +1,45 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.ScheduleCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Date;
+import seedu.address.model.person.SchedulePredicate;
+
+/**
+ * Parses input arguments and creates a new ScheduleCommand object
+ */
+public class ScheduleCommandParser implements Parser<ScheduleCommand> {
+    /**
+     * Parses {@code userInput} into a command and returns it.
+     *
+     * @param args
+     * @throws ParseException if {@code userInput} does not conform the expected format
+     */
+    @Override
+    public ScheduleCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+              ArgumentTokenizer.tokenize(args, PREFIX_DATE);
+
+        //at least one feature is present
+        if (!arePrefixesPresent(argMultimap, PREFIX_DATE) || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ScheduleCommand.MESSAGE_USAGE));
+        }
+        //can only see schedule for one day at a time
+        if (argMultimap.getAllValues(PREFIX_DATE).size() > 1) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ScheduleCommand.MESSAGE_USAGE));
+        }
+
+        Date date = new Date(argMultimap.getValue(PREFIX_DATE).orElse(""));
```

Comment:
Consider adding a validation that ensures the date falls within a reasonable range